CurrentLocationKotlin/app/src/main/java/com/example/android/location/currentlocationkotlin/MainActivity.kt

```diff
@@ -15,12 +15,183 @@
  */
 package com.example.android.location.currentlocationkotlin
 
-import androidx.appcompat.app.AppCompatActivity
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.location.Location
+import android.net.Uri
 import android.os.Bundle
+import android.provider.Settings
+import android.util.Log
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.example.android.location.currentlocationkotlin.databinding.ActivityMainBinding
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationRequest.PRIORITY_HIGH_ACCURACY
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.tasks.CancellationTokenSource
+import com.google.android.gms.tasks.Task
+import com.google.android.material.snackbar.Snackbar
 
+/**
+ * Demonstrates how to easily get the current location via the [FusedLocationProviderClient.getCurrentLocation].
+ * The main code is in this class's requestCurrentLocation() method.
+ */
 class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+
+    // The Fused Location Provider provides access to location APIs.
+    private val fusedLocationClient: FusedLocationProviderClient by lazy {
+        LocationServices.getFusedLocationProviderClient(applicationContext)
+    }
+
+    // Allows class to cancel the location request if it exits the activity.
+    // Typically, you use one cancellation source per lifecycle.
+    private var cancellationTokenSource = CancellationTokenSource()
+
+    // If the user denied a previous permission request, but didn't check "Don't ask again", this
+    // Snackbar provides an explanation for why user should approve, i.e., the additional rationale.
+    private val fineLocationRationalSnackbar by lazy {
+        Snackbar.make(
+            binding.container,
```

Comment:
Probably not critical since it's not a fragment, but I was curious that doesn't this leak the view of the Activity?